### PR TITLE
Fix for wrong endian determination when signature is same for both

### PR DESCRIPTION
**Pull request by richkong88**
endians.

### CODE REVIEW

File: library/spdm_crypt_lib/libspdm_crypt_asym.c [hunk 1]
```diff
@@ -979,10 +1004,12 @@ bool libspdm_asym_verify_ex(
         }
     }
     if (try_big_endian && try_little_endian && result) {
-        if (little_endian_succeeded) {
-            *endian = LIBSPDM_SPDM_10_11_VERIFY_SIGNATURE_ENDIAN_LITTLE_ONLY;
-        } else {
-            *endian = LIBSPDM_SPDM_10_11_VERIFY_SIGNATURE_ENDIAN_BIG_ONLY;
+        if (!libspdm_is_palindrome(signature, sig_size)) {
```
Comment (jyao1) [inline library/spdm_crypt_lib/libspdm_crypt_asym.c:1007]: This is for whole signature, right?

RSA is OK.

But ECDSA includes 2 parts - R and S.
Should we test R and S separately?

Comment (richkong88) [reply library/spdm_crypt_lib/libspdm_crypt_asym.c:1007]: You are correct. I forgot.

Comment (richkong88) [reply library/spdm_crypt_lib/libspdm_crypt_asym.c]: @jyao1 I updated the changes for your review.

File: library/spdm_crypt_lib/libspdm_crypt_asym.c [hunk 2]
```diff
@@ -689,6 +689,64 @@ void libspdm_copy_signature_swap_endian(
...
+
+    const uint32_t spdm_10_11_ecdsa_algos =
+        SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P256 |
+        SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P384 |
+        SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P521;
+
+    if (base_asym_algo & spdm_10_11_rsa_algos) {
+        return libspdm_is_palindrome(buf, buf_size);
+    } else if (base_asym_algo & spdm_10_11_ecdsa_algos) {
+        size_t x_size = buf_size / 2;
```
Comment (steven-bellock) [inline library/spdm_crypt_lib/libspdm_crypt_asym.c:736]: `x_size` and `y_size` can be `const` as well.

**Review by jyao1 — APPROVED** [review]

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**